File: 2017/day1-1.c
#include <stdio.h>

int main(void) {
  // Load number
  int numbers[2042];
  // from file
  FILE *myFile;
  myFile = fopen("day1.txt", "r");
  for (int i=0; i<2042; i++)
  {
    fscanf(myFile, "%1d", &numbers[i]);
  }

  // Check for repeated values and sum
  int sum = 0;
  for (int i=0; i<2041; i++) {
    if (numbers[i] == numbers[i+1]) {
      sum += numbers[i];
    }
  }
  // last digit with first digit
  if (numbers[2041] == numbers[0]) {
    sum += numbers[0];
  }

  printf("Solution = %d\n", sum);
  return 0;
}

File: 2020/day1-1.py
with open("day1.txt", "r") as f:
    data = [int(number[:-1]) for number in f.readlines()]

for i, n1 in enumerate(data):
    for j, n2 in enumerate(data[i:]):
        if n1 + n2 == 2020:
            print(n1 * n2)
